kornia/augmentation/random_generator/types.py

```diff
@@ -0,0 +1,13 @@
+from typing import Dict, Optional
+
+import torch
+try:
+    from typing import TypedDict # type: ignore  # >=3.8  
+except ImportError:
+    from mypy_extensions import TypedDict  # type: ignore  # <=3.7
+
+
+class AugParamDict(TypedDict, total=False):  # type: ignore
```

Comment:
First, I would rename to `AugmentationParameters` to make it more clear since it's not derived from dict. Second, take look at python `dataclasses`, I think it's supported in python >3.6 , double check if there's any workaround on this. Last, why do you `type: ignore` ? this might hurt jitting which is something that we should start to implement.

Reply:
Renaming is reasonable. ```TypedDict``` will be compiled as normal ```dict``` at the runtime, so it will not cause any issues there. While the ```type: ignore``` is a versioning issue that ```total=False``` is only supported in 3.8 or so.

The reason that I did not go for another class is that I am not sure if we wish to make user construct params like ```from kornia.augmentation import AugmentationParams; param=AugmentationParams(height=1, width=1)```. Would it be overcomplicated?